warehouse/ingest-core/src/main/java/datawave/ingest/table/aggregator/GlobalIndexUidAggregator.java

```diff
@@ -174,59 +174,9 @@ public Value reduce(Key key, Iterator<Value> iter) {
-                        // for a given key and therefore there's no need to include removed UIDs in the
-                        // output protocol buffer since the remove have already been applied.
-                        if (timestampsIgnored) {
-                            uids.remove(uid);
-                            // Even if propagate is false, the removal UID should persist until all PB lists
-                            // in the current iteration have been seen, in case a subsequent PB has the UID
-                            // marked for removal.
-                            uidsToRemove.add(uid);
-                        } else if (propogate && !uids.contains(uid)) {
-                            uidsToRemove.add(uid);
-                        }
-                    }
-                    
-                    // Add UIDs from the UID list
-                    for (String uid : v.getUIDList()) {
-                        // Don't add a uid that's been removed. This is the same whether or
-                        // not timestamps are ignored since if they are ignored, removals take
-                        // priority and if they are not ignored, then this add is happening
-                        // before a removal and therefore should not take place.
-                        if (!uidsToRemove.contains(uid)) {
-                            // Add the UID iff we are under our MAX. If we reach the max,
-                            // then treat it as though we've seen an ignore--don't try to
-                            // add any more UIDs to the list (or removals from the REMOVEDUIDs
-                            // list) since they won't be included when we aggregate anyway.
-                            if (uids.size() < maxUids) {
-                                uids.add(uid);
-                            } else if (!uids.contains(uid)) {
-                                // This UID will not push the PB over the max UID limit if it is
-                                // already in the list of UIDs.
-                                // If aggregating this PB pushed us over the max UID limit,
-                                // then ignore any work we've done integrating this PB so far
-                                // and instead treat it as though its ignore flag had been set.
-                                // Set the count to the number of collected UIDs before we went
-                                // over the max, and then we'll add this PB's count below.
-                                seenIgnore = true;
-                                count = prevCount;
-                                uids.clear();
-                                uidsToRemove.clear();
-                                break;
-                            }
-                        }
+                    boolean wasSuccessful = processRemovalUids(v) && processAddedUids(v);
```

Comment:
nice, I like the simplification here.

Reply:
Instead of `wasSuccessful`, maybe we can call this `exceededMaxThreshold` or something like that? `wasSuccessful` sorta makes you think the operation can fail.

Reply:
Good idea, thanks.